### PR TITLE
Add translation debug configuration

### PR DESCRIPTION
**Pull request by cchabilall83**
This allows developers to see if the content strings have been in translated in their respective locales.  Changes in the localApiController check if the en _localkey_ exists in the enabled local packs, if not add it, this allow for the tx-source pack to be read as the default en.

Steps to enable

1. Symlink tx-source dir from the locales repo to your vanilla/locales dir
2.  Add following keys to your config
`$Configuration['EnabledLocales']['tx-source'] = 'en';`
`$Configuration['TranslationDebug'] = true;`

requires https://github.com/vanilla/locales/pull/237

## Other Notes

- Static calls to `t()` have never worked. They now have a warning if they are called before the translations are setup. All existing calls have been updated.
- The test config had to be updated to properly configure the test environmental variable.

### CODE REVIEW

File: packages/vanilla-i18n/src/translationStore.ts [hunk 1]
```diff
@@ -41,6 +59,11 @@ export function translate(str: string, defaultTranslation?: string): string {
         return translationStore[str];
     }
 
+    // report any untranslated strings when in translation debug mode
+    if (defaultTranslation === undefined && translationDebug(getMeta("context.translationDebug"))) {
```
Comment (charrondev) [inline packages/vanilla-i18n/src/translationStore.ts]: Please use the `internalTranslationDebugValue` here. These packages should never have a dependency on library.

File: packages/vanilla-i18n/src/translationStore.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import { getMeta } from "@vanilla/library/src/scripts/utility/appUtils";
```
Comment (charrondev) [inline packages/vanilla-i18n/src/translationStore.ts]: This import should be removed. See below.

File: library/src/scripts/bootstrap.ts [hunk 3]
```diff
@@ -20,6 +21,9 @@ import { bootstrapLocales } from "@library/locales/localeBootstrap";
 const debugValue = getMeta("context.debug", getMeta("debug", false));
 debug(debugValue);
 
+const translationDebugValue = getMeta("context.translationDebug", getMeta("translationDebug", false));
```
Comment (charrondev) [inline library/src/scripts/bootstrap.ts]: What's the reason for the nesting fallback? This value should always be in the same place.

File: packages/vanilla-i18n/src/translationStore.ts (hunk 1 above)
Comment (cchabilall83) [reply packages/vanilla-i18n/src/translationStore.ts]: removed the getMeta call

File: packages/vanilla-i18n/src/translationStore.ts (hunk 2 above)
Comment (cchabilall83) [reply packages/vanilla-i18n/src/translationStore.ts]: done

File: library/src/scripts/bootstrap.ts (hunk 3 above)
Comment (cchabilall83) [reply library/src/scripts/bootstrap.ts]: removed second getMeta call

File: packages/vanilla-i18n/src/translationStore.ts [hunk 4]
```diff
@@ -41,6 +57,11 @@ export function translate(str: string, defaultTranslation?: string): string {
         return translationStore[str];
     }
 
+    // report any untranslated strings when in translation debug mode
+    if (defaultTranslation === undefined && translationDebug()) {
```
Comment (charrondev) [inline packages/vanilla-i18n/src/translationStore.ts]: I'd even go a bit further. I think we should wrap the result even with the "default" provided.

```js
    let result = defaultTranslation !== undefined ? defaultTranslation : str;

    if (translationDebug()) {
    	result = "☢️☢️☢️" + result + "☢️☢️☢️";
    }
	return result;
```

**Pull request merged**